### DISCUSSION TITLE
NUMBER_OF_PROCESSORS not set on the windows workers

### DISCUSSION THREAD

**Discussion by jwillemsen** [community/community#25133] [Actions]
<p>Would it be possible to set NUMBER_OF_PROCESSORS on the Windows workers? I noticed it is not set because our builds takes long, this is normally set by Windows for the amount of processors available.</p>

**Comment by weide-zhou** [comment]
<p>Hi <a class="mention" href="/jwillemsen">@jwillemsen</a> ,</p>
<p>GitHub hosts Windows runners on <a href="https://docs.microsoft.com/en-us/azure/virtual-machines/dv2-dsv2-series#dsv2-series" rel="nofollow noopener">Standard_DS2_v2</a> virtual machines in Microsoft Azure, more compute options may be added in the future but currently it’s not supported to set the number of the processors.</p>
<p>The first offering here will be the ability to provide <a href="https://help.github.com/en/actions/getting-started-with-github-actions/core-concepts-for-github-actions#self-hosted-runner" rel="nofollow noopener">self-hosted</a> runners, so you can bring your own compute needs to GitHub Actions.  </p>
<p>Thanks.</p>

**Comment by jwillemsen** [comment]
<p>It would help when NUMBER_OF_PROCESSORS would be set by GitHub/azure, so that we can retrieve the number of processors available in our script</p>

**Comment by weide-zhou** [comment]
<p>Thanks for your reply! <a class="mention" href="/jwillemsen">@jwillemsen</a>. According to the policy, it’s recommened to raise a ticket <a href="https://support.github.com/contact/feedback?contact%5Bcategory%5D=actions" rel="nofollow noopener">here</a> where github product manager will take a review.</p>

**Comment by MSP-Greg** [comment, marked as answer]
<p>I’m a bit confused.  If you run the following:</p>
<pre><code>gci env:* | sort-object name
</code></pre>
<p>NUMBER_OF_PROCESSORS is set to 2?</p>
<p>See any Windows job at <a href="https://github.com/MSP-Greg/actions-image-testing/actions" rel="nofollow noopener">https://github.com/MSP-Greg/actions-image-testing/actions</a></p>

**Comment by weide-zhou** [comment]
<p>Thanks <a class="mention" href="/msp-greg">@msp-greg</a> , you’re right, the ‘NUMBER_OF_PROCESSORS’ can be got from env.</p>
<p>And the value can be set via ‘set-env’ command, code sample as below:</p>
<pre><code>- name: set Number of processors
        run: |
          echo "::set-env name=NUMBER_OF_PROCESSORS::1"
</code></pre>
<p>Hi <a class="mention" href="/jwillemsen">@jwillemsen</a> , sorry i misunderstood. No error reported if you change the value, but since the hosted runner is a '<a href="https://docs.microsoft.com/en-us/azure/virtual-machines/dv2-dsv2-series#dsv2-series" rel="nofollow noopener">Standard_DS2_v2</a> virtual machines in Microsoft Azure, i’m not sure if it works for you, could you please kindly check? Thanks.</p>

**Comment by jwillemsen** [comment]
<p>Thanks, I will retest, one remark, instead of powershell I am using cmd as shell.</p>

**Comment by jwillemsen** [comment]
<p>Can confirm that it is set, didn’t check the logs in enough detail</p>
